src/equations/compressible_euler_2d.jl

```diff
@@ -1263,4 +1263,43 @@ end
 end
 
 
+# Calculate the vorticity on a single node using the derivative matrix from the polynomial basis of
+# a DGSEM solver.
+# !!! Note that this only works for DGSEM solvers on a TreeMesh !!!
+# `u` is the solution on the whole domain.
+# This function is used for calculating acoustic source terms for coupled Euler-acoustics
+# simulations.
+function calc_vorticity_node(u, equations::CompressibleEulerEquations2D, dgsem, cache, i, j, element)
+  @unpack derivative_matrix = dgsem.basis
+
+  v2_x = zero(eltype(u)) # derivative of v2 in x direction
+  for ii in eachnode(dgsem)
+    u_node = get_node_vars(u, equations, dgsem, ii, j, element)
+    v2 = u_node[3] / u_node[1]
```

Comment:
```suggestion
    rho, _, rho_v2 = get_node_vars(u, equations, dgsem, ii, j, element)
    v2 = rho_v2 / rho
```
I usually prefer making this explicit instead of requiring people reading the code to remember the order of all variables... Applies also to the loop below